cognite/client/data_classes/data_modeling/data_types.py

```diff
@@ -125,23 +125,58 @@ class Json(ListablePropertyType):
     _type = "json"
 
 
+@dataclass(frozen=True)
+class UnitReference:
+    external_id: str
+    source_unit: str | None = None
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = {"externalId" if camel_case else "external_id": self.external_id}
+        if self.source_unit:
+            output["sourceUnit" if camel_case else "source_unit"] = self.source_unit
+        return output
+
+    @classmethod
+    def load(cls, data: dict) -> UnitReference:
+        return cls(
+            external_id=data["externalId"],
+            source_unit=data.get("sourceUnit"),
+        )
+
+
+@dataclass(frozen=True)
+class UnitSystemReference:
+    unit_system_name: str
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {"unitSystemName" if camel_case else "unit_system_name": self.unit_system_name}
+
+    @classmethod
+    def load(cls, data: dict) -> UnitSystemReference:
+        return cls(unit_system_name=data["unitSystemName"])
+
+
 @dataclass
 class ListablePropertyTypeWithUnit(ListablePropertyType, LoadablePropertyType, ABC):
-    unit: NodeId | None = None
+    unit: UnitReference | None = None
```

Comment:
Breaking change?

Reply:
Looked into this. This was from the last implementation of Units which was quickly removed. So I would say that this is technically a bug now which has been fixed by setting it correctly.

Reply:
sgtm.